### PR TITLE
Make the environment (dev/prod) a parameter #170

### LINKED ISSUE #170: make the environment (dev/prod) a parameter
useful for testing

### PR DESCRIPTION
**Pull request by lyomr**
This PR:

- checks if environment variables flags are set
- Makes a parameter called nodeEnv
- add param to package.json
- Closes #170

### CODE REVIEW

File: roosevelt.js [hunk 1]
```diff
@@ -10,15 +10,18 @@ var http = require('http'),
...
   params = params || {}; // ensure params are an object
 
-  // check for command line overrides for NODE_ENV
+  //flag for env variables
+  var isEnvFlagSet = false;
   process.argv.forEach(function (val, index, array) {
     switch (val) {
       case '-dev':
         process.env.NODE_ENV = 'development';
+        isEnvFlagSet = true;
```
Comment (kethinov) [inline roosevelt.js]: instead of defining a local variable, just set the param

File: roosevelt.js [hunk 2]
```diff
@@ -44,6 +47,19 @@ module.exports = function(params) {
 
   console.log(('💭  ' + 'Starting ' + app.get('appName') + ' in ' + app.get('env') + ' mode...').bold);
 
+  // check if env comes from cmd line then leave as is, else use params
```
Comment (kethinov) [inline roosevelt.js]: once the param is set by the command line flag, you can just move this whole block into sourceParams, where it belongs

Comment (kethinov) [inline roosevelt.js]: also make sure to tweak the code changes to the build passes

Comment (alallier) [inline roosevelt.js]: I'm confused here. The point of the param is to handle the case when you don't have a command line flag to set the environment

File: roosevelt.js (hunk 1 above)
Comment (alallier) [inline roosevelt.js]: Similar comment to below

File: roosevelt.js (hunk 2 above)
Comment (kethinov) [inline roosevelt.js]: as i understand it the hierarchy of expected behavior is, in order of priority:

1. command line flag for dev/prod (currently available, triggered by `npm run dev` and `run prod` commands)
2. constructor param (currently unavailable, this PR implements this, intended to supersede `NODE_ENV`, but not command line flags)
3. package.json param  (currently unavailable, this PR implements this, intended to supersede `NODE_ENV`, but not command line flags)
4. `NODE_ENV` environment variable (currently available, intended to be what decides this if no command line flags or params are supplied)

File: lib/sourceParams.js [hunk 3]
```diff
@@ -40,6 +40,7 @@ module.exports = function(app) {
     shutdownTimeout: params.shutdownTimeout || pkg.shutdownTimeout || 30000,
     bodyParserUrlencodedParams: params.bodyParserUrlencodedParams || pkg.rooseveltConfig.bodyParserUrlencodedParams || { extended: true },
     bodyParserJsonParams: params.bodyParserJsonParams || pkg.rooseveltConfig.bodyParserJsonParams || {},
+    devProd: params.devProd || pkg.rooseveltConfig.devProd,
```
Comment (kethinov) [inline lib/sourceParams.js]: rename this to `nodeEnv`

File: roosevelt.js [hunk 4]
```diff
@@ -10,15 +10,16 @@ var http = require('http'),
 module.exports = function(params) {
   params = params || {}; // ensure params are an object
 
-  // check for command line overrides for NODE_ENV
```
Comment (kethinov) [inline roosevelt.js]: no reason to remove this comment

**Comment by kethinov** [comment]
document this param in the README in the appropriate places

File: README.md [hunk 5]
```diff
@@ -327,7 +327,8 @@ Statics parameters
   - Example: `library-name/example.js:lib/example.min.js` (customizes both file path and file name of minified file)
 - `jsCompiledOutput`: Where to place compiled JS files. This folder will be made public by default.
   - Default: `.build/js`
-
+- `nodeEnv`: Sets environment variable in development or production mode. Default is handled by the script
```
Comment (kethinov) [inline README.md]: change this to `nodeEnv`: Param to override the `NODE_ENV` environment variable.

File: README.md [hunk 6]
```diff
@@ -327,7 +327,8 @@ Statics parameters
...
+  - Default: `dev or prod`
```
Comment (kethinov) [inline README.md]: default should be `undefined`

File: lib/sourceParams.js [hunk 7]
```diff
@@ -40,6 +40,7 @@ module.exports = function(app) {
     shutdownTimeout: params.shutdownTimeout || pkg.shutdownTimeout || 30000,
     bodyParserUrlencodedParams: params.bodyParserUrlencodedParams || pkg.rooseveltConfig.bodyParserUrlencodedParams || { extended: true },
     bodyParserJsonParams: params.bodyParserJsonParams || pkg.rooseveltConfig.bodyParserJsonParams || {},
+    nodeEnv: params.nodeEnv || pkg.rooseveltConfig.nodeEnv,
```
Comment (kethinov) [inline lib/sourceParams.js]: if neither the param nor the rooseveltConfig supplies `nodeEnv`, it should default to `process.env.NODE_ENV`

File: README.md [hunk 8]
```diff
@@ -327,7 +327,8 @@ Statics parameters
   - Example: `library-name/example.js:lib/example.min.js` (customizes both file path and file name of minified file)
 - `jsCompiledOutput`: Where to place compiled JS files. This folder will be made public by default.
   - Default: `.build/js`
-
+- `nodeEnv`: Param to override the `NODE_ENV environment` variable.
```
Comment (kethinov) [inline README.md]: `NODE_ENV` not `NODE_ENV environment`

Comment (lyomr) [inline README.md]: I missed taking that out

File: roosevelt.js [hunk 9]
```diff
@@ -15,10 +15,12 @@ module.exports = function(params) {
     switch (val) {
       case '-dev':
         process.env.NODE_ENV = 'development';
+        params.nodeEnv = 'dev';
```
Comment (kethinov) [inline roosevelt.js]: match the express values: `development` and `production`

**Pull request merged**